src/components/wizards/create-vm/CreateVMWizard.js

```diff
@@ -0,0 +1,366 @@
+import React from 'react';
+import { Form, FormControl, Col, ButtonGroup, DropdownButton, MenuItem, Checkbox, FormGroup, ControlLabel, HelpBlock, Wizard } from 'patternfly-react';
+import './CreateVMWizard.css';
+import * as _ from 'lodash';
+import { StorageTable } from './StorageTable';
+
+const defaultState = {
+  activeStepIndex: 0,
+  basicSettings: {},
+  wizardValid: false
+}
+
+export class CreateVMWizard extends React.Component {
+
+    constructor(props){
+      super(props);
+      this.state = defaultState;
+    }
+  
+    onFormChange = (newValue, target) => {
+      let valid = null;
+      if (this.basicSettings[target].required && newValue.trim().length === 0){
+        valid= `${this.basicSettings[target].title} is required`;
+      }
+      const basicSettings = {
+        ...this.state.basicSettings,
+        [target]: {
+          value: newValue,
+          valid: valid
+        }
+      };
+      this.setState({
+        basicSettings: basicSettings
+      });
+      this.validateWizard(basicSettings);
```

Comment:
At this point, the component has already re-rendered, since `this.setState` triggers a new render.

`this.validateWizard` calls `this.setState` again, triggering another re-render.

We can improve this by doing validation as part of component's render.